src/client/core/utils/shared/visibility.ts

```diff
@@ -31,13 +42,15 @@ export function isElementVisible (el: Node): boolean {
     }
 
     if (adapter.dom.isSVGElement(el)) {
-        if (adapter.style.get(el, 'visibility') === 'hidden' || adapter.style.get(el, 'display') === 'none')
-            return false;
+        const hiddenParent = adapter.dom.findParent(el, true, (parent: Node) => {
+            return hiddenUsingStyles(parent as unknown as HTMLElement);
+        });
 
-        const parent = adapter.dom.findParent(el);
+        if (!hiddenParent)
+            return !hiddenByRectangle(el as unknown as HTMLElement);
 
-        return parent ? isElementVisible(parent) : true;
+        return false;
```

Comment:
It's also may be easier
```suggestion
        return !hiddenParent && !hiddenByRectangle(el as unknown as HTMLElement);
```

Reply:
I disagreed with this remark. It is more difficult to read then origin variant.